src/mlpack/methods/logistic_regression/logistic_regression_function_impl.hpp

```diff
@@ -171,6 +206,29 @@ void LogisticRegressionFunction<MatType>::Gradient(
       * (responses[i] - sigmoid) + regularization;
 }
 
+//! Evaluate the gradient of the logistic regression
+//  objective function for a given batch size.
+template<typename MatType>
+void LogisticRegressionFunction<MatType>::Gradient(
+                const arma::mat& parameters,
+                const size_t begin,
+                const size_t batchSize,
+                arma::mat& gradient) const
+{
+  // Regularization term.
+  arma::mat regularization;
+  regularization = lambda * parameters.col(0).subvec(1, parameters.n_elem - 1);
+
+  const arma::rowvec sigmoids = (1 / (1 + arma::exp(-parameters(0, 0)
+      - parameters.col(0).subvec(1, parameters.n_elem - 1).t() * predictors)));
+
+  gradient.set_size(batchSize);
+  gradient[0] = -arma::accu(responses - sigmoids);
+  gradient.col(0).subvec(begin, batchSize - 1) = -predictors * (responses -
```

Comment:
Here again, we could use the slice of predictors that the batch requires predictors.col(begin, begin + batchSize)